packages/nodes-base/nodes/HttpRequest/GenericFunctions.ts

```diff
@@ -138,21 +138,31 @@ export const binaryContentTypes = [
 export type BodyParametersReducer = (
 	acc: IDataObject,
 	cur: { name: string; value: string },
-) => IDataObject;
+) => Promise<IDataObject>;
 
-export const prepareRequestBody = (
+export async function reduceAsync<T, R>(
+	arr: T[],
+	reducer: (acc: Awaited<Promise<R>>, cur: T) => Promise<R>,
+	init: Promise<R> = Promise.resolve({} as R),
+): Promise<R> {
+	return arr.reduce(async (promiseAcc, item) => {
+		return reducer(await promiseAcc, item);
+	}, init);
+}
```

Comment:
Decided to add a adapter in the interest of time and to minimize the refactor, but generally speaking I find the heavy use of reducers in the HTTP node rather hard to follow.